62.CAD/common.hlsl

```diff
@@ -29,49 +40,58 @@ struct DrawObject
 struct QuadraticBezierInfo
 {
     float64_t2 p[3]; // 16*3=48bytes
-    // TODO[Przemek]: Any Data related to precomputing things for beziers will be here
+    float64_t2 arcLen;
 };
 
-// TODO[Lucas]:
-/*
-You need a struct here that represents a curve which is referenced by a "CurveBox"
-Which is basically the same as QuadraticBezierInfo, if the middle point is "nan" it means it's a line connected by p0 and p2
-*/
-
-// TODO[Lucas]:
-/*
-You need another struct here that represents a "CurveBox" which
-0. have a aabb (double2 min, max) which will get transformed in the vertex shader, and will be calculated on the cpu when generating these boxes
-1. references two Curves by `uint64_t address` into the geometry buffer
-2. It will also contain tmin,tmax for both curves (becuase we subdivide curves into smaller monotonic parts we need this info to help us discard invalid solutions)
-*/
+struct CurveBox 
+{
+    // will get transformed in the vertex shader, and will be calculated on the cpu when generating these boxes
+    float64_t2 aabbMin;
+    float64_t2 aabbMax; // 32
+    float64_t2 curveMin[3]; // 80
+    float64_t2 curveMax[3]; // 128
+};
 
 // TODO: Compute this in a compute shader from the world counterparts
 //      because this struct includes NDC coordinates, the values will change based camera zoom and move
 //      of course we could have the clip values to be in world units and also the matrix to transform to world instead of ndc but that requires extra computations(matrix multiplications) per vertex
 struct ClipProjectionData
 {
-    float64_t3x3 projectionToNDC; // 72 -> because we use scalar_layout
-    float32_t2 minClipNDC; // 80
-    float32_t2 maxClipNDC; // 88
+    float64_t4x4 projectionToNDC; // 128 -> because we use scalar_layout
```

Comment:
why did you change back to float4x4?! turn it into float3x3 and adjust your other code